src/main/java/htsjdk/samtools/CigarOperator.java

```diff
@@ -28,40 +28,62 @@
  */
 public enum CigarOperator {
     /** Match or mismatch */
-    M(true, true,   'M'),
+    M(true, true),
     /** Insertion vs. the reference. */
-    I(true, false,  'I'),
+    I(true, false),
     /** Deletion vs. the reference. */
-    D(false, true,  'D'),
+    D(false, true),
     /** Skipped region from the reference. */
-    N(false, true,  'N'),
+    N(false, true),
     /** Soft clip. */
-    S(true, false,  'S'),
+    S(true, false),
     /** Hard clip. */
-    H(false, false, 'H'),
+    H(false, false),
     /** Padding. */
-    P(false, false, 'P'),
+    P(false, false),
```

Comment:
I would just have 1 constructor that takes the  character and the binary representation as parameters.  It's a bit redundant but it seems clearer to me.